src/main/java/frc/robot/subsystems/gripper/Gripper.java

```diff
@@ -0,0 +1,98 @@
+package frc.robot.subsystems.gripper;
+
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.units.Voltage;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+public class Gripper extends SubsystemBase {
+    private static Gripper INSTANCE;
+    private final GripperIO io;
+    public GripperIO.ControlMode controlMode;
+    private final GripperInputsAutoLogged inputs = GripperIO.inputs;
+
+    @AutoLogOutput private final Mechanism2d mechanism2d = new Mechanism2d(0, 0);
+
+    private final MechanismRoot2d root = mechanism2d.getRoot("Gripper", 0, 0);
+    private final MechanismLigament2d gripper =
+            root.append(new MechanismLigament2d("Gripper", 0, 0));
+
+    private Gripper(GripperIO io) {
+        this.io = io;
+    }
+
+    public static Gripper getInstance() {
+        return INSTANCE;
+    }
+
+    public static void initialize(GripperIO io) {
+        INSTANCE = new Gripper(io);
+    }
+
+    public void setRollerMotorPower(double power) {
+        io.setRollerMotorPower(power);
+        controlMode = GripperIO.ControlMode.PERCENT_OUTPUT;
+    }
+
+    public void setAngleMotorPower(double power) {
+        io.setAngleMotorPower(power);
+        controlMode = GripperIO.ControlMode.PERCENT_OUTPUT;
+    }
+
+    public void setAngle(MutableMeasure<Angle> angle) {
+        io.setAngle(angle);
+        controlMode = GripperIO.ControlMode.POSITION;
+    }
+
+    public MutableMeasure<Angle> getCurrentAngle() {
+        return inputs.currentAngle;
+    }
+
+    public MutableMeasure<Angle> getAngleSetpoint() {
+        return inputs.angleSetpoint;
+    }
+
+    public boolean hasNote() {
+        return inputs.hasNote;
+    }
+
+    public MutableMeasure<Voltage> getRollerMotorVoltage() {
+        return inputs.rollerMotorVoltage;
+    }
+
+    public MutableMeasure<Voltage> getAngleMotorVoltage() {
+        return inputs.angleMotorVoltage;
+    }
+
+    public Command setRollerPower(double power) {
+        return run(() -> io.setRollerMotorPower(power));
+    }
+
+    public Command intake() {
+        return run(() -> io.setAngle(GripperConstants.INTAKE_ANGLE))
+                .alongWith(run(()->io.setRollerMotorPower(GripperConstants.INTAKE_POWER)));
+    }
+    public Command outtake() {
+        return run(() -> io.setAngle(GripperConstants.OUTTAKE_ANGLE))
+                .alongWith(run(()->io.setRollerMotorPower(GripperConstants.OUTTAKE_POWER)));
+    }
+
+    public Command setWristPosition(MutableMeasure<Angle> angle) {
+        return run(() -> io.setAngle(angle));
+    }
```

Comment:
Forgot to comment that on the elevator too -- you need to use the `withName()` decorator. Otherwise, the name of the command will be e.g. `ParallelCommandGroup`.

Reply:
d02a228ce282deb1e543cab9de7e271a218c697f

Reply:
1943b2706203c8a576bced6c58b04d25464d360b